### PR TITLE
Stream Search Functionality on home page

### PR DESCRIPTION
**Pull request by praveen5959**

### CODE REVIEW

File: src/pages/Home/index.tsx [hunk 1]
```diff
@@ -68,15 +82,39 @@ const Home: FC = () => {
 	const [userSpecificStreams, setAppStore] = useAppStore((store) => store.userSpecificStreams);
 	const [userRoles] = useAppStore((store) => store.userRoles);
 	const [userAccessMap] = useAppStore((store) => store.userAccessMap);
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredMetaData, setFilteredMetaData] = useState<Record<string, any>>(() => metaData || {});
```
Comment (Koustavd18) [inline src/pages/Home/index.tsx:86]: Type should not be any

**Review by Koustavd18 — CHANGES_REQUESTED** [review]

File: src/pages/Home/index.tsx (hunk 1 above)
Comment (praveen5959) [reply src/pages/Home/index.tsx]: Addressed this

File: src/pages/Home/index.tsx [hunk 2]
```diff
@@ -68,15 +84,42 @@ const Home: FC = () => {
...
+		}
+	}, [metaData]);
 
 	useEffect(() => {
 		if (!Array.isArray(userSpecificStreams) || userSpecificStreams.length === 0) return;
 		getStreamMetadata(userSpecificStreams.map((stream) => stream.name));
 	}, [userSpecificStreams]);
 
+	const filterStreams = useCallback(
+		debounce((term: string) => {
```
Comment (Koustavd18) [inline src/pages/Home/index.tsx:103]: we should not debounce on data that already there in memory

**Review by Koustavd18 — CHANGES_REQUESTED** [review]

File: src/pages/Home/index.tsx (hunk 2 above)
Comment (praveen5959) [reply src/pages/Home/index.tsx]: Addressed this

**Review by nikhilsinhaparseable — APPROVED** [review]
good to merge

**Pull request merged**